packages/dotcom-ui-layout/src/components/__test__/Layout.spec.tsx

```diff
@@ -1,227 +1,233 @@
 /**
  * @jest-environment jsdom
  */
-import 'jest-enzyme'
 import React from 'react'
-import { shallow } from 'enzyme'
+import { render } from '@testing-library/react'
 
 import { Layout as Subject } from '../Layout'
-import { Header, Drawer, LogoOnly, NoOutboundLinksHeader } from '@financial-times/dotcom-ui-header'
-import { Footer, LegalFooter } from '@financial-times/dotcom-ui-footer'
 
-describe('dotcom-ui-layout/src/components/Layout', () => {
-  const result = shallow(<Subject />)
+jest.mock('@financial-times/dotcom-ui-header', () => ({
+  Header: () => <div>Header mock</div>,
+  Drawer: () => <div>Drawer mock</div>,
+  LogoOnly: () => <div>LogoOnly mock</div>,
+  NoOutboundLinksHeader: () => <div>NoOutboundLinksHeader mock</div>
+}))
+
+jest.mock('@financial-times/dotcom-ui-footer', () => ({
+  Footer: () => <div>Footer mock</div>,
+  LegalFooter: () => <div>LegalFooter mock</div>
+}))
 
+describe('dotcom-ui-layout/src/components/Layout', () => {
   it('renders skip links to page landmarks', () => {
-    expect(result.find('a[href="#site-navigation"]')).toHaveText('Skip to navigation')
-    expect(result.find('a[href="#site-content"]')).toHaveText('Skip to content')
-    expect(result.find('a[href="#site-footer"]')).toHaveText('Skip to footer')
-  })
+    const { container } = render(<Subject />)
+
+    const skipToNavigationLink = container.querySelector('a[href="#site-navigation"]')
+    const skipToContentLink = container.querySelector('a[href="#site-content"]')
+    const skipToFooterLink = container.querySelector('a[href="#site-footer"]')
 
-  it('renders the header and drawer components', () => {
-    expect(result.find(Header)).toExist()
-    expect(result.find(Drawer)).toExist()
```

Comment:
**question:** What I like about the previous approach is that we could also query what properties were passed to the mock. The new method we lose that as we have completely mocked the component. Are we able to keep the previous method of shallow rendering the child components without mocking them completely?

Reply:
With mocks in react testing library it's still possible to check the props passed to the children! https://stackoverflow.com/a/58634377

Secondly, it seems shallow rendering as a concept has died out with enzyme. I can't find it in the new react docs, just the legacy ones. I'd love to be proven wrong on this one however, if you know any more on this topic. A good article for why shallow is flawed if you're interested: https://kentcdodds.com/blog/why-i-never-use-shallow-rendering